powerauth-java-crypto/src/test/java/io/getlime/security/powerauth/crypto/encryption/GeneralEncryptorTest.java

```diff
@@ -0,0 +1,959 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2023 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.security.powerauth.crypto.encryption;
+
+import io.getlime.security.powerauth.crypto.lib.encryptor.EncryptorFactory;
+import io.getlime.security.powerauth.crypto.lib.encryptor.RequestResponseValidator;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ServerEncryptor;
+import io.getlime.security.powerauth.crypto.lib.encryptor.ClientEncryptor;
+import io.getlime.security.powerauth.crypto.lib.encryptor.exception.EncryptorException;
+import io.getlime.security.powerauth.crypto.lib.encryptor.model.*;
+import io.getlime.security.powerauth.crypto.lib.encryptor.model.v3.ClientEncryptorSecrets;
+import io.getlime.security.powerauth.crypto.lib.encryptor.model.v3.ServerEncryptorSecrets;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.util.ByteUtils;
+import io.getlime.security.powerauth.crypto.lib.util.KeyConvertor;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.UUID;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test to validate functionality of high level {@link ServerEncryptor}
+ * and {@link ClientEncryptor} classes.
+ */
+public class GeneralEncryptorTest {
+
+    private final static EncryptorFactory encryptorFactory = new EncryptorFactory();
+    private final static KeyGenerator keyGenerator = new KeyGenerator();
+    private final static KeyConvertor keyConvertor = new KeyConvertor();
+
+    private final static List<EncryptorId> encryptorIds = List.of(
+            EncryptorId.APPLICATION_SCOPE_GENERIC,
+            EncryptorId.ACTIVATION_SCOPE_GENERIC,
+            EncryptorId.ACTIVATION_LAYER_2,
+            EncryptorId.CREATE_TOKEN,
+            EncryptorId.VAULT_UNLOCK,
+            EncryptorId.UPGRADE,
+            EncryptorId.CONFIRM_RECOVERY_CODE
+    );
+
+    private String APPLICATION_KEY;
```

Comment:
It does not comply with Java conventions. It should be either lower-case or _final static_.

Reply:
I got inspiration from original ECIES test. Such values are typically constants hardcoded in the mobile app (e.g. final static), and that's the reason why it break rules. From the test-case perspective, the value doesn't change during the one particular test. We don't want then static and final, because it's good to use a different values for each test.